lib/rmt/mirror.rb

```diff
@@ -85,7 +85,8 @@ def create_temp_dir
     raise RMT::Mirror::Exception.new(_('Could not create a temporary directory: %{error}') % { error: e.message })
   end
 
-  def mirror_metadata(repository_dir, repository_url, temp_metadata_dir, n_retry = MIRROR_METADATA_RETRIES)
+  def mirror_metadata(repository_dir, repository_url, temp_metadata_dir)
+    attempts ||= MIRROR_METADATA_RETRIES
```

Comment:
I'd suggest to call this variable `remaining_attempts` to make clear that this is the remaining amount of retries. 
The constant `MIRROR_METADATA_RETRIES` could be called `MIRROR_RETRIES` because it's not only about metadata anymore.